### DISCUSSION TITLE
Missing `ref struct` document

### DISCUSSION THREAD

**Discussion by raffaeler** [dotnet/csharplang#2023] [Q&A] [answered]
I could not find the specification or proposal document for `ref struct`.
The only document mentioning it is the one on [span safety](
https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md) but it does not contain all the details required to call it a spec.
In addition to that, I found a [blog post](https://blogs.msdn.microsoft.com/mazhou/2018/03/02/c-7-series-part-9-ref-structs/) where it is stated that a `ref struct` *represents a sequential struct layout*. But this concept is not detailed in any document in this repo or the official documentation.
More generally, I agree with [this request](https://github.com/dotnet/csharplang/issues/64) about the official document specification for the current released language version.

**Comment by Zenexer** [comment]
I'm not sure if this is quite what you're looking for, but just in case:

* [Sequential struct layout](https://docs.microsoft.com/en-us/dotnet/framework/interop/default-marshaling-behavior#default-marshaling-for-value-types)
* [Ref structs](https://github.com/dotnet/docs/blob/master/docs/csharp/write-safe-efficient-code.md)

**Comment by raffaeler** [comment]
Hi @Zenexer the struct layout document refers to the main (old) documentation about that feature. My request is about the new language specs about `ref struct`.
None of the two documents tells anything about that.
Thanks anyway.

**Comment by hypeartist** [comment, marked as answer]
@raffaeler https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/keywords/ref.md#ref-struct-types
